api/krusty/extendedpatch_test.go

```diff
@@ -1213,3 +1066,44 @@ spec:
     app: busybox
 `)
 }
+
+func TestTargetMissingPatchJson6902Error(t *testing.T) {
+	th := kusttest_test.MakeHarness(t)
+	makeCommonFileForExtendedPatchTest(th)
+	th.WriteK("base", `
+resources:
+- servicemonitor.yaml
+patchesJson6902:
+- target:
+    group: monitoring.coreos.com
+    kind: ServiceMonitor
+    name: starboard-exporter
+    namespace: starboard
+    version: v2
+  path: patch.0.yaml
+`)
+	th.WriteF("base/servicemonitor.yaml", `
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  labels:
+    app: starboard-exporter
+  name: starboard-exporter
+  namespace: starboard
+spec:
+  endpoints:
+  - path: /metrics
+    port: metrics
+  selector:
+    matchLabels:
+      app.kubernetes.io/instance: starboard-exporter
+      app.kubernetes.io/name: starboard-exporter
+`)
+	th.WriteF("base/patch.0.yaml", `
+- op: add
+  path: /metadata/labels/release
+  value: kube-prometheus-stack
+`)
+	err := th.RunWithErr("base", th.MakeDefaultOptions())
+	assert.Contains(t, err.Error(), "patchesJson6902 target not found for ServiceMonitor.v2.monitoring.coreos.com/starboard-exporter.starboard:a=:l=t.")
```

Comment:
What is the `:a=:l=t.` at the end? Can we remove it from the error message?

Reply:
sure, I will add only Target's name in the error message.

Reply:
@natasha41575 - Updated PR with the requested changes. PTAL.

Reply:
> Just a quick FYI: the a stands for annotationSelector and l for labelSelector that can be specified in target

Thanks for the explanation! Still probably not something we want to surface to users in such an incomprehensible way, so I'd prefer the error just include GVKN/N.

Reply:
@natasha41575 @annasong20 - Updated PR as per the above discussion. PTAL.